packages/modular-scripts/src/__tests__/test.test.ts

```diff
@@ -194,6 +194,209 @@ describe('Modular test command', () => {
       );
     });
   });
+
+  describe('test command can successfully do selective tests based on selected packages', () => {
+    const fixturesFolder = path.join(
+      __dirname,
+      Array.from({ length: 4 }).reduce<string>(
+        (acc) => `${acc}..${path.sep}`,
+        '',
+      ),
+      '__fixtures__',
+      'ghost-testing',
+    );
+
+    const currentModularFolder = getModularRoot();
+    let randomOutputFolder: string;
+
+    beforeEach(() => {
+      // Create random dir
+      randomOutputFolder = tmp.dirSync({ unsafeCleanup: true }).name;
+      fs.copySync(fixturesFolder, randomOutputFolder);
+      execa.sync('yarn', {
+        cwd: randomOutputFolder,
+      });
+    });
+
+    // Run in a single test, serially for performance reasons (the setup time is quite long)
+    it('finds --package after specifying a valid workspaces / finds ancestors using --ancestors', () => {
+      const resultPackages = runRemoteModularTest(
+        currentModularFolder,
+        randomOutputFolder,
+        ['test', '--package', 'b', '--package', 'c'],
+      );
+      expect(resultPackages.stderr).toContain(
+        'packages/c/src/__tests__/utils/c-nested.test.ts',
+      );
+      expect(resultPackages.stderr).toContain(
+        'packages/c/src/__tests__/c.test.ts',
+      );
+      expect(resultPackages.stderr).toContain(
+        'packages/b/src/__tests__/utils/b-nested.test.ts',
+      );
+      expect(resultPackages.stderr).toContain(
+        'packages/b/src/__tests__/b.test.ts',
+      );
+
+      const resultPackagesWithAncestors = runRemoteModularTest(
+        currentModularFolder,
+        randomOutputFolder,
+        ['test', '--ancestors', '--package', 'b', '--package', 'c'],
+      );
+      expect(resultPackagesWithAncestors.stderr).toContain(
```

Comment:
nit: could maybe use a single `expect` statement here

Reply:
Unless you're saying that we can conflate `packages/c/src/__tests__/utils/c-nested.test.ts` and `packages/c/src/__tests__/utils/c-nested.test.ts` into their common root: `packages/c/src/__tests__/`; in that case I explicitly wanted to show that we're executing all the tests of a workspace, including those in subdirectories.

Reply:
What's the shape of `resultPackagesWithAncestors.stderr`? I can imagine we can find a way to dedupe this a bit, but it's really not a big deal